docs/stack/client/querying.md

```diff
@@ -0,0 +1,164 @@
+
+    const gql = require('graphql-tag')   
+
+    // titles of all proposals of the DAO
+    let query = gql`query {
+      proposals ( where: { dao: "0x294f999356ed03347c7a23bcbcf8d33fa41dc830" }) { title }
+    }`
+
+### Executing Query
+
+As we noted above the query created need to be explicitly sent to the server.
+
+#### Entity Methods
+By default `subscribe` parameter is set to `false`. This is useful to send a one time query and not bother the server with further requests for subscription.
+
+    // Get all proposals' Id of the DAO
+    dao.proposals().subscribe(() => {})
+
+#### Raw GraphQL queries
+These queries are made using the static method `arc.sendQuery`. Pass the query designed [above](#raw-graphql-queries) as the parameter to `sendQuery`.
+
+    // Get votes id and outcome of the given Proposal
+    arc.sendQuery(gql`query {
+        proposal (id: "0x1245") {
+          votes { id outcome }
+        }}`)
+
+
+### Subscribing Query
+*Subscriptions* are used so that our handler gets called every time a new value is emitted for the observable stream. This is useful to keep the app data updated as the value changes.
+
+#### Entity methods
+As we saw the Entity methods do not send the query to the server but return an observable. We must subscribe as follows to send the query as well as subscribe for updates.
+
+```
+arc.daos({subscribe: true}).subscribe(() => {})
+dao.state({subscribe: true}).subscribe( () => {})
+```
+  
+#### Raw graphQL queries
+
+For even more control over what data is being fetched and subscribed to, you can write explicit queries:
+
+```
+arc.subscribe(gql`subscribe { proposal (id: "0x1245") { votes { id outcome }})`)
+new Proposal("0x1245").votes().subscribe(
+  (next) => {
+
+  }
+)
+```
+
+## Optimizing Subscription and Using Cache
+Since, subscriptions can be expensive, this behavior can be controlled/optimized in several ways.
+  
+  - [Controlling fetchPolicy](#controlling-apollo-fetchpolicy): controlling cache interaction.
+  - [Subscribing to Apollo Cache](#subscribe-to-apollo-cache-changes): getting updates from Apollo cache instead of graphnode server.
+  - [FetchAllData and Nested subscription](#use-fetchalldata-with-nested-subscription): by querying larger set at top level and subscribing to Apollo cache for nested queries.
+
+### Controlling Apollo fetchPolicy
+We can pass Apollo's [`fetchPolicy`](https://www.apollographql.com/docs/react/api/react-apollo/#optionsfetchpolicy) argument to control how the query interacts with the cache:
+
+  - _**cache-first**_: default value. Read data from cache first, fetch from network otherwise.
+  - _**cache-and-network**_: return data from cache but still always fetch from network. It optimizes quick response while also keep cached data updated.
```

Comment:
I don't understand cache-and-network from this explanation.  Can this be clarified?  "but still always" means what?